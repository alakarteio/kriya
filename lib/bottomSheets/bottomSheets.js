'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; };

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _propTypes = require('prop-types');

var _propTypes2 = _interopRequireDefault(_propTypes);

var _classnames2 = require('classnames');

var _classnames3 = _interopRequireDefault(_classnames2);

var _recompose = require('recompose');

var _bottomSheetsStyles = require('../../src/bottomSheets/bottomSheets.styles.scss');

var _bottomSheetsStyles2 = _interopRequireDefault(_bottomSheetsStyles);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _defineProperty(obj, key, value) { if (key in obj) { Object.defineProperty(obj, key, { value: value, enumerable: true, configurable: true, writable: true }); } else { obj[key] = value; } return obj; }

var BottomSheets = function BottomSheets(_ref) {
  var style = _ref.style,
      className = _ref.className,
      children = _ref.children,
      toggle = _ref.toggle,
      print = _ref.print,
      header = _ref.header,
      contents = _ref.contents;

  var classes = (0, _classnames3.default)(_bottomSheetsStyles2.default.bottomSheets, className);

  var classSheets = (0, _classnames3.default)(_bottomSheetsStyles2.default.sheets, _defineProperty({}, _bottomSheetsStyles2.default.print, print));

  return (
    // eslint-disable-next-line jsx-a11y/no-static-element-interactions
    _react2.default.createElement(
      'div',
      { style: style, className: classes },
      _react2.default.createElement(
        'div',
        { className: classSheets },
        _react2.default.createElement(
          'div',
          { className: _bottomSheetsStyles2.default.header, onClick: toggle },
          header
        ),
        _react2.default.createElement('hr', { className: _bottomSheetsStyles2.default.separator }),
        _react2.default.createElement(
          'div',
          { className: _bottomSheetsStyles2.default.content },
          _react2.default.cloneElement(children, _extends({ print: print }, contents))
        )
      )
    )
  );
};

BottomSheets.propTypes = {
  style: _propTypes2.default.object,
  className: _propTypes2.default.string,
  children: _propTypes2.default.node.isRequired,
  toggle: _propTypes2.default.func,
  print: _propTypes2.default.bool,
  header: _propTypes2.default.object,
  contents: _propTypes2.default.object
};

BottomSheets.defaultProps = {
  style: {},
  className: '',
  toggle: undefined,
  print: false,
  header: undefined,
  contents: undefined
};

exports.default = (0, _recompose.onlyUpdateForPropTypes)(BottomSheets);