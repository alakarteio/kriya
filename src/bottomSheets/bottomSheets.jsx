import React from 'react'
import PropTypes from 'prop-types'
import classnames from 'classnames'
import { onlyUpdateForPropTypes } from 'recompose'
import styles from '../../src/bottomSheets/bottomSheets.styles.scss'

const BottomSheets = ({ style, className, children, toggle, print, header, contents }) => {
  const classes = classnames(
    styles.bottomSheets,
    className,
  )

  const classSheets = classnames(
    styles.sheets,
    {
      [styles.print]: print,
    },
  )

  return (
    // eslint-disable-next-line jsx-a11y/no-static-element-interactions
    <div style={style} className={classes}>
      <div className={classSheets}>
        <div className={styles.header} onClick={toggle}>{header}</div>
        <hr className={styles.separator} />
        <div className={styles.content}>
          {React.cloneElement(children, { print, ...contents })}
        </div>
      </div>
    </div>
  )
}

BottomSheets.propTypes = {
  style: PropTypes.object,
  className: PropTypes.string,
  children: PropTypes.node.isRequired,
  toggle: PropTypes.func,
  print: PropTypes.bool,
  header: PropTypes.object,
  contents: PropTypes.object,
}

BottomSheets.defaultProps = {
  style: {},
  className: '',
  toggle: undefined,
  print: false,
  header: undefined,
  contents: undefined,
}

export default onlyUpdateForPropTypes(BottomSheets)
