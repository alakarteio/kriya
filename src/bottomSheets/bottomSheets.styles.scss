@import '~styles/theme';

$header-height: 2em;

.bottomSheets {
  bottom: 0;
  left: 0;
  margin: 0 2em;
  position: fixed;
  right: 0;
}

.sheets {
  align-items: center;
  background-color: $default-bg-color;
  box-shadow: 0 0 .2em 0 $font-color-dark;
  display: flex;
  flex-direction: column;
  margin: 0 auto;
  transform: translateY(calc(100% - #{$header-height}));
  transition: transform .2s;

  &.print {
    transform: translateY(0);
  }

}

.header {
  align-items: center;
  cursor: pointer;
  display: flex;
  height: $header-height;
  width: 100%;
}

.content {
  background-color: $default-bg-color;
  justify-content: center;
}

.separator {
  border: 0;
  border-bottom: solid 1px $default-grey-color;
  width: 100%;
}
