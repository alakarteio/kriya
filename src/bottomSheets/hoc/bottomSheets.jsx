import React from 'react'
import BottomSheets from '../bottomSheets.container'

export default config => Component => props => (
  <BottomSheets {...config}>
    <Component {...props} />
  </BottomSheets>
)
