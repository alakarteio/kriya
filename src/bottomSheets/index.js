export { default } from './bottomSheets.container'
export * from './redux'
export { default as reducer } from './redux'
export { default as popover } from './hoc'
