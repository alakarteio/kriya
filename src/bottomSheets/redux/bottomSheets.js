import { OPEN_BOTTOMSHEETS, CLOSE_BOTTOMSHEETS, TOGGLE_BOTTOMSHEETS } from './bottomSheets.actions'

export const initState = { }
export const initAction = { type: 'UNKNOWN' }

export default (state = initState, { type, payload } = initAction) => {
  switch (type) {
    case OPEN_BOTTOMSHEETS:
      return { ...state, [payload.code]: { print: true, contents: payload.contents } }
    case CLOSE_BOTTOMSHEETS:
      return { ...state, [payload]: { ...state[payload], print: false } }
    case TOGGLE_BOTTOMSHEETS:
      return {
        ...state,
        [payload]: { ...state[payload], print: !(state[payload] && state[payload].print) },
      }
    default: return state
  }
}
