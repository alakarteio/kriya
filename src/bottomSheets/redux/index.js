import bottomSheets from './bottomSheets'
import * as actions from './bottomSheets.actions'
import * as selectors from './bottomSheets.selectors'

Object.assign(bottomSheets, actions, selectors)

export * from './bottomSheets.actions'
export * from './bottomSheets.selectors'
export default bottomSheets
