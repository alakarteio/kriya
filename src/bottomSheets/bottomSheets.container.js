import { connect } from 'react-redux'
import { toggle, get } from './redux'
import Component from './bottomSheets'

const mapStateToProps = (state, { code }) => {
  const bottomSheets = get(state)[code]
  if (!bottomSheets) return {}
  return bottomSheets
}

const mapDispatchToProps = (dispatch, { closable, code }) => {
  if (!closable) return {}

  return {
    toggle: () => dispatch(toggle(code)),
  }
}

export default connect(mapStateToProps, mapDispatchToProps)(Component)
