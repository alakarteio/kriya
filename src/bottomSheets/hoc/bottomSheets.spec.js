/* eslint-disable
  import/no-extraneous-dependencies,
  react/jsx-filename-extension,
  react/no-children-prop,
  react/prop-types,
*/
/* eslint-env jest */

import React from 'react'
import snap from 'snap'
import bottomSheets from './bottomSheets'

jest.mock('../bottomSheets.container', () => ({ children, ...props }) => <div>{JSON.stringify(props)}-{children}</div>)
const Child = props => <div title="child">{JSON.stringify(props)}</div>

describe('common/BottomSheets/hoc', () => {
  it('should pass config to the bottomSheets component', () => {
    const Wrapper = bottomSheets({ code: 'a code' })(Child)
    snap(Wrapper)()()
  })

  it('should pass props to the wrapped component', () => {
    const Wrapper = bottomSheets()(Child)
    snap(Wrapper)({ prop: 'a prop', and: 'an other' })()
  })
})

/* eslint-enable
  import/no-extraneous-dependencies,
  react/jsx-filename-extension,
  react/no-children-prop,
  react/prop-types,
*/
