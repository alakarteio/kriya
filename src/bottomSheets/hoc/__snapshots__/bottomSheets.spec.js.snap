// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`common/BottomSheets/hoc should pass config to the bottomSheets component 1`] = `
<div>
  {"code":"a code"}
  -
  <div
    title="child"
  >
    {}
  </div>
</div>
`;

exports[`common/BottomSheets/hoc should pass props to the wrapped component 1`] = `
<div>
  {}
  -
  <div
    title="child"
  >
    {"prop":"a prop","and":"an other"}
  </div>
</div>
`;
